geode-for-redis/src/main/java/org/apache/geode/redis/internal/commands/RedisCommandType.java

```diff
@@ -164,214 +166,293 @@
-  ZREM(new ZRemExecutor(), SUPPORTED, new Parameter().min(3).flags(WRITE, FAST)),
-  ZREMRANGEBYLEX(new ZRemRangeByLexExecutor(), SUPPORTED, new Parameter().exact(4).flags(WRITE)),
-  ZREMRANGEBYRANK(new ZRemRangeByRankExecutor(), SUPPORTED, new Parameter().exact(4).flags(WRITE)),
-  ZREMRANGEBYSCORE(new ZRemRangeByScoreExecutor(), SUPPORTED,
+  ZRANGEBYLEX(new ZRangeByLexExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().min(4).flags(READONLY)),
+  ZRANGEBYSCORE(new ZRangeByScoreExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().min(4).flags(READONLY)),
+  ZRANK(new ZRankExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().exact(3).flags(READONLY, FAST)),
+  ZREM(new ZRemExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().min(3).flags(WRITE, FAST)),
+  ZREMRANGEBYLEX(new ZRemRangeByLexExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().exact(4).flags(WRITE)),
+  ZREMRANGEBYRANK(new ZRemRangeByRankExecutor(), Category.SORTEDSET, SUPPORTED,
       new Parameter().exact(4).flags(WRITE)),
-  ZREVRANGE(new ZRevRangeExecutor(), SUPPORTED,
+  ZREMRANGEBYSCORE(new ZRemRangeByScoreExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().exact(4).flags(WRITE)),
+  ZREVRANGE(new ZRevRangeExecutor(), Category.SORTEDSET, SUPPORTED,
       new Parameter().min(4).max(5, ERROR_SYNTAX).flags(READONLY)),
-  ZREVRANGEBYLEX(new ZRevRangeByLexExecutor(), SUPPORTED, new Parameter().min(4).flags(READONLY)),
-  ZREVRANGEBYSCORE(new ZRevRangeByScoreExecutor(), SUPPORTED,
+  ZREVRANGEBYLEX(new ZRevRangeByLexExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().min(4).flags(READONLY)),
+  ZREVRANGEBYSCORE(new ZRevRangeByScoreExecutor(), Category.SORTEDSET, SUPPORTED,
       new Parameter().min(4).flags(READONLY)),
-  ZREVRANK(new ZRevRankExecutor(), SUPPORTED, new Parameter().exact(3).flags(READONLY, FAST)),
-  ZSCAN(new ZScanExecutor(), SUPPORTED, new Parameter().min(3).flags(READONLY, RANDOM),
+  ZREVRANK(new ZRevRankExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().exact(3).flags(READONLY, FAST)),
+  ZSCAN(new ZScanExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().min(3).flags(READONLY, RANDOM),
       new Parameter().odd(ERROR_SYNTAX)),
-  ZSCORE(new ZScoreExecutor(), SUPPORTED, new Parameter().exact(3).flags(READONLY, FAST)),
-  ZUNIONSTORE(new ZUnionStoreExecutor(), SUPPORTED,
+  ZSCORE(new ZScoreExecutor(), Category.SORTEDSET, SUPPORTED,
+      new Parameter().exact(3).flags(READONLY, FAST)),
+  ZUNIONSTORE(new ZUnionStoreExecutor(), Category.SORTEDSET, SUPPORTED,
       new Parameter().min(4).flags(WRITE, DENYOOM, MOVABLEKEYS)),
 
   /************* Server *****************/
-  COMMAND(new COMMANDCommandExecutor(), SUPPORTED, new Parameter().min(1).firstKey(0).flags(RANDOM,
+  COMMAND(new CommandCommandExecutor(), Category.CONNECTION, SUPPORTED, new Parameter().min(1).firstKey(0).flags(RANDOM,
```

Comment:
With the addition of these categories, does it make sense to group categories together in this file? We currently have a section for `CONNECTION` but not for `ADMIN` or `KEYSPACE`, and there is no `SERVER` category, so having everything agree on what it is/where it goes would probably make the file a little neater. This isn't that important though.

Reply:
That's a good suggestion.